Quick/Nimble
# Add throwAssertion matcher

Fixes #234. Adds on @abbeycode's work in #248. Changes on top of that PR:

- Update CwlPreconditionTesting for Swift 3
- Add error message tests for `throwAssertion`
- Disabled debugging of Nimble's tvOS test bundle to run `throwAssertion` tests
- Add warning for first usage of `throwAssertion` in tvOS to remind to disable debugging for it to work correctly.

What's left?

- [x] Update Documentation examples (see tests)
- [x] Document using `throwAssertion` matcher for tvOS

Changes to also do at @mattgallagher's CwlPreconditionTesting:

- [x] Send upstream PR for CwlPreconditionTesting for POSIX's Swift 3 support
- [x] Send upstream PR for CwlPreconditionTesting to fix https://github.com/mattgallagher/CwlPreconditionTesting/issues/5
